docs/src/main/sphinx/security/authentication-types.rst

```diff
@@ -20,6 +20,7 @@ and authenticators are available:
 * ``KERBEROS`` for :doc:`kerberos`
 * ``CERTIFICATE`` for :doc:`certificate`
 * ``JWT`` for :doc:`jwt`
+* ``HEADER-BASED`` for :doc:`/develop/header-based-authenticator` (SPI only)
```

Comment:
Let's remove the "SPI only" note

I'm thinking we could shorten the name to `HEADER`, since it seems a bit redundant and verbose, and maybe also drop "based" everywhere. That fits better with the rest of them being single word names.